fix: return unique_list only when link_extractor.unique is True

reidemeister94 opened the pull request:
In the `lxmlhtml.py` script, the function `extract_links(self, response)` always returns `unique_list(all_links)`, even when the attribute `unique` of the `link_extractor` is equal to `False`. 

The behavior is coherent with the `unique` attribute with this little change in the return line.

Fixes https://github.com/scrapy/scrapy/issues/3798, closes https://github.com/scrapy/scrapy/pull/3799, closes https://github.com/scrapy/scrapy/pull/4695

wRAR commented:
See also: #3798 #4695 #3799

reidemeister94 commented:
Ok so this won't get fixed?

reidemeister94 commented:
Any update on this?

Gallaecio commented:
Added to my review queue, but it may take a while :bow:

reidemeister94 commented:
Just a ping :D

Gallaecio commented:
I am OK with merging as is, provided we update the docstring of `extract_links` and provide a test for the change.

reidemeister94 commented:
I updated the docstring and fixed the html used for the tests to have tests that also cover the new case (unique = false, and the output contains duplicates). 
I hope that's enough for being merged :D

Gallaecio reviewed (approved):
Until your tests changes I did not realize the issue only affected links with matching URL *and* text, but the original issue indeed describes it like that.

Thanks!

Gallaecio commented on scrapy/linkextractors/lxmlhtml.py:
```diff
@@ -195,7 +195,8 @@ def extract_links(self, response):
         the link extractor are returned.
 
-        Duplicate links are omitted.
+        Duplicate links are omitted if the ``unique`` parameter is set to ``True``, 
```
```suggestion
        Duplicate links are omitted if the ``unique`` attribute is set to ``True``, 
```
